app/src/main/java/com/calculate/CalcActivity.java

```diff
@@ -1,20 +1,436 @@
+                            mResult = first * second;
+                            break;
+                        case '/':
+                            if (mSecond.equals("0")) {
+                                throw new ArithmeticException("Деление на 0");
+                            }
+                            mResult = first / second;
+                            break;
+                        default:
+                            mResult = 0;
+                    }
+                    resultText = "" + mResult;
+                    if (resultText.equals("Infinity")) {
+                        throw new ArithmeticException("Переполнение");
+                    }
+                    resultText = splitZero(resultText);
+
+                }
+
+                addCharToText(resultText);
+            } catch (Exception ex) {
+                addCharToText(" : " + ex.getMessage());
+                mOperation = '!';
+            }
+
+
+        }
+    };
+
+    /**
+     * Метод удаления 0, точек и запятых
+     *
+     * @param resultText - строка на обработку
+     * @return обработанная строка
+     */
+    private String splitZero(String resultText) {
+        boolean exist0 = false;
+        if (resultText.contains(".") || resultText.contains(",")) {
+            exist0 = true;
+        }
+        while (exist0) {
+            if (resultText.charAt(resultText.length() - 1) == '0' && resultText.length() != 1) {
+                resultText = resultText.substring(0, resultText.length() - 1);
+            } else if (resultText.charAt(resultText.length() - 1) == ',' || resultText.charAt(resultText.length() - 1) == '.') {
+                resultText = resultText.substring(0, resultText.length() - 1);
+                exist0 = false;
+            } else {
+                exist0 = false;
+            }
+        }
+        return resultText;
+    }
+
+    /**
+     * Метод добавления символа к параметру
+     *
+     * @param key - добавляемый символ
+     */
+    private void addCharToParam(String key) {
+        if (tvResultText.contains("=")) {
+            initCalc();
+        }
+        if (mOperation == '!') {
+            return;
+        } else if (mOperation == ' ') {
+            if ((!key.equals(".") || !mFirst.contains(".")) && mFirst.length()<15) {
+                if (mFirst.length() == 0 && key.equals(".")) key = "0" + key;
+                mFirst += key;
+                addCharToText(key);
+            }
+        } else {
+            if ((!key.equals(".") || !mSecond.contains(".")) && mSecond.length()<15) {
+                if (mSecond.length() == 0 && key.equals(".")) key = "0" + key;
+                mSecond += key;
+                addCharToText(key);
+            }
+        }
+
+
+    }
+
+    /**
+     * Метод инициализации результирующего поля
+     *
+     * @param key - добавляемый символ
+     */
+    private void addCharToText(String key) {
+
+        if (tvResult.getText() == "0") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.